### DISCUSSION TITLE
Error in linux install

### DISCUSSION THREAD

**Discussion by FKLCo** [fra589/cn5X#34] [Q&A] [answered]
hallo sir
I install cn5X without problem, but when I try to run I receive the follwing error:
catalin@catalin-ThinkPad-X201:~$ '/home/catalin/Descărcări/cn5X-master/cn5X.py' 
qt5ct: using qt5ct plugin
cn5X++ v0.8.6 running from: /home/catalin/Descărcări/cn5X-master

                ####### #     #
  ####   #    # #        #   #     #       #
 #    #  ##   # #         # #      #       #
 #       # #  #  #####     #     #####   #####
 #       #  # #       #   # #      #       #
 #    #  #   ## #     #  #   #     #       #
  ####   #    #  #####  #     #

qt5ct: D-Bus global menu: no
Traceback (most recent call last):
  File "/home/catalin/Descărcări/cn5X-master/cn5X.py", line 2757, in <module>
    window = winMain()
  File "/home/catalin/Descărcări/cn5X-master/cn5X.py", line 170, in __init__
    self.setTranslator(langue)
  File "/home/catalin/Descărcări/cn5X-master/cn5X.py", line 2681, in setTranslator
    self.log(logSeverity.error.value, self.tr("Locale ({}) not usable, using default to english").format(locale.name()))
AttributeError: module 'locale' has no attribute 'name'

Linux Mint 20 Ulyana pe 64 de biți
Linux 5.4.0-96-generic x86_64
Default computer language - Romanian
Thanks for any advice.

**Comment by fra589** [comment, marked as answer]
Hi @FKLCo,  
  
Thanks for using cn5X and report this.  
You found a new bug!  
I created this issue from here: https://github.com/fra589/cn5X/issues/35, correction will be pushed in few minutes :-)  
  
@++;  
Gauthier.

**Reply by fra589** [reply]
Correction pushed now in new 0.8.7 version.

**Reply by FKLCo** [reply]
Thanks sir! Works now.

**Reply by FKLCo** [reply]
And for installation, I have to install:
sudo apt-get install python3-pyqt5.qtmultimedia

**Reply by fra589** [reply]
You are right ! It's due to the last update I made... I'm currently adding sound feedback according to certain actions (for example during probe contacts) and I haven't yet put update the readme.

**Comment by FKLCo** [comment]
Sir 
I play with cn5X and I liked but maybe I do something wrong or the program is in beta stage!?
Please see the picture
![Poaza 1](https://user-images.githubusercontent.com/80265277/151716163-0fb73fdf-2b91-4325-ac0f-79e5bad10764.png)

For G1 I set F500 and is reported 180?
The motor run like 180.

![Poza 2](https://user-images.githubusercontent.com/80265277/151716173-4ac0cd50-8dac-46bb-9266-4638c4abeaff.png)

Almost same problem for S value.

![Poza 3](https://user-images.githubusercontent.com/80265277/151716174-96d91f23-dfd1-4c39-94ef-af0794c6db5a.png)
.
The buffer never exceed  34 value for short Gcode, and stay at 0 alost all the time for long Gcode (stay at 0 for G0 and G1 execution period and jump to 34 at M5, after that drop to 0 again).

I dont know if this is a bug or I do something wrong.

![Poza 4](https://user-images.githubusercontent.com/80265277/151716629-a65b0ab8-17c9-4f46-a6be-c7e53ec2e3c2.png)

For test I use this Gcode
[Test Gcode.zip](https://github.com/fra589/cn5X/files/7966875/Test.Gcode.zip)

Thanks sir.

**Reply by fra589** [reply]
Hi @FKLCo,  
  
In your cn5X interface pictures, I can see that you have set Feedrate and Spindle overlay to 36%. So, the difference you see is normal.  
For the buffer status, this mean that when you use your short GCode file, there is not enough lines to fill the buffer. You also have a G4 (Dwell) command which cause to wait for the end of movement then pause for a time. Waiting to the end of movement has the effect of emptying the buffer.  
When you use your large file, this mean than the buffer is regularly filled, this mean that your Arduino card is able to calculate movement more rapidly than the time needed to doing them.  
  
@++;  
Gauthier.

**Reply by FKLCo** [reply]
Uh, sorry I dont see the S and F is set to 36%. sorry again.

**Comment by FKLCo** [comment]
Sir 
I don't fund-it or do not exist the key shortcut for a functions?
I want to use a numeric keypad for regular task.
I put a link for better understanding.
https://create.arduino.cc/projecthub/edr1924/arduino-hid-based-cnc-pendant-aefa13 
thanks.

**Reply by fra589** [reply]
Hi @FKLCo,  
  
It's in my TODO list to add device interface like this CNC pendant to cn5X...  
But I don't know when I will have time to this :flushed: :disappointed_relieved: :snail:  
:clock1::clock2::clock3::clock4::clock5::clock6::clock7::clock8::clock9::clock10::clock11::clock12:  
  
But in my point of view, simulate keyboard shortcuts is not the best option. I think I will use an USB/Serial, Bluetooth or WiFi protocol for communication between devices:
\<CNC Pendant>---\<cn5X>---\<grbl[-Mega-5X]>  
This protocol need to be defined, and perhaps it will be compatible with a direct communication between the CNC pendant and Grbl itself with a new grbl-Mega-5X interface definition added to Grbl's software.
  
For the moment, there is no shortcut implemented in cn5X.  
  
@++;
Gauthier.

**Comment by jcrogomez** [comment]
Hi, Im having a similar issue, suddenly the bottom line FS:0,24000 doesn't relate to the GUI Federate but the spindle speed is ok, I have the overlays at 100 percent
